extras/reps_and_likelihoods.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019-2022 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+//
+// The purpose of this code is documented in the header.
+
+#include "reps_and_likelihoods.hpp"
+
+int main(int argc, char *argv[]) {
+  if (argc != 8) {
+    std::cout
+        << "We need exactly 7 arguments: fasta, rooted_nwk, credible_rooted_nwk,"
+           "pp_rooted_nwk, repr_out_path, credible_repr_out_path, and pp_repr_out_path"
+        << std::endl;
+    abort();
+  }
+  std::string fasta_path = argv[1];
+  std::string rooted_nwk_path = argv[2];
+  std::string credible_rooted_nwk_path = argv[3];
+  std::string pp_rooted_nwk_path = argv[4];
+  std::string out_path = argv[5];
+  std::string credible_out_path = argv[6];
+  std::string pp_out_path = argv[7];
+  auto thread_count = std::thread::hardware_concurrency();
+
+  GPInstance all_trees_gp_inst("mmapped_plv.data");
+  RootedSBNInstance cred_r_inst("cred_trees");
+  RootedSBNInstance pp_r_inst("pp_trees");
+  all_trees_gp_inst.ReadNewickFile(rooted_nwk_path);
+  all_trees_gp_inst.ReadFastaFile(fasta_path);
+  all_trees_gp_inst.MakeEngine();
+  all_trees_gp_inst.TakeFirstBranchLength();
+  cred_r_inst.ReadNewickFile(credible_rooted_nwk_path);
+  pp_r_inst.ReadNewickFile(pp_rooted_nwk_path);
+
+  auto all_trees = all_trees_gp_inst.GenerateCompleteRootedTreeCollection();
+  auto cred_trees = cred_r_inst.tree_collection_;
+  auto pp_trees = pp_r_inst.tree_collection_;
+  auto indexer = all_trees_gp_inst.GetDAG().BuildGPCSPIndexer();
+  //std::vector<RootedIndexerRepresentation> all_representations;
```

Comment:
The new version looks great, though can you clear out these comments please?

Reply:
Oops. Yes, fixed.